pkg/controllers/rollout/controller.go

```diff
@@ -418,20 +368,146 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 	// the list of bindings that are to be updated by this rolling phase
 	toBeUpdatedBindingList := make([]toBeUpdatedBinding, 0)
 	if len(removeCandidates)+len(updateCandidates)+len(boundingCandidates)+len(applyFailedUpdateCandidates) == 0 {
-		return toBeUpdatedBindingList, nil, false, nil
+		return toBeUpdatedBindingList, nil, false, minWaitTime, nil
 	}
 
-	// calculate the max number of bindings that can be unavailable according to user specified maxUnavailable
-	maxUnavailableNumber, _ := intstr.GetScaledValueFromIntOrPercent(crp.Spec.Strategy.RollingUpdate.MaxUnavailable, targetNumber, true)
-	minAvailableNumber := targetNumber - maxUnavailableNumber
-	// This is the lower bound of the number of bindings that can be available during the rolling update
-	// Since we can't predict the number of bindings that can be unavailable after they are applied, we don't take them into account
-	lowerBoundAvailableNumber := len(readyBindings) - len(canBeUnavailableBindings)
-	maxNumberToRemove := lowerBoundAvailableNumber - minAvailableNumber
-	klog.V(2).InfoS("Calculated the max number of bindings to remove", "clusterResourcePlacement", crpKObj,
-		"maxUnavailableNumber", maxUnavailableNumber, "minAvailableNumber", minAvailableNumber,
-		"lowerBoundAvailableBindings", lowerBoundAvailableNumber, "maxNumberOfBindingsToRemove", maxNumberToRemove)
+	toBeUpdatedBindingList, staleUnselectedBinding := determineBindingsToUpdate(crp, removeCandidates, updateCandidates, boundingCandidates, applyFailedUpdateCandidates, targetNumber,
+		readyBindings, canBeReadyBindings, canBeUnavailableBindings, crpKObj)
+
+	return toBeUpdatedBindingList, staleUnselectedBinding, true, minWaitTime, nil
+}
+
+// Process unscheduled bindings
+func (r *Reconciler) processUnscheduledBinding(
+	crpKObj, bindingKObj klog.ObjectRef,
```

Comment:
I thought about refactoring the func but the parameters are so huge so I gave up. I guess it's still worth it.

However,  we can get rid of ObjectRef. 

We should also make sure what are the input vs output